lib/OpenQA/WebAPI/Plugin/Helpers.pm

```diff
@@ -57,7 +57,12 @@ sub register {
     $app->helper(
         bugicon_for => sub {
             my ($c, $text) = @_;
-            return ($text =~ /(poo|gh)#/) ? 'label_bug fa fa-bolt' : 'label_bug fa fa-bug';
+            my $css_class = ($text =~ /(poo|gh)#/) ? 'label_bug fa fa-bolt' : 'label_bug fa fa-bug';
+            my $bug = OpenQA::Schema::Result::Bugs->get_bug($text, $c->db);
```

Comment:
this is heavy. This comes down to a SQL query per icon - in a view. There must be better ways to gather the information right in the controller, where you can also make use of proper joins and caching